### PR TITLE
Add notes about SQL-only builder to README [HZ-1874]

### PR DESCRIPTION
**Pull request by sumnerib**
Adds `run-sql-only` to the list of "Trigger Phrases" as well as a more complete description of the job and corresponding label.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -353,10 +353,17 @@ see used in the comments on your PR:
   `extensions/cdc-postgres` module
 * `run-s3-tests` - run all tests in the `extensions/s3` module
 * *`run-nightly-tests` - run nightly (slow) tests. WARNING: Use with care as this is a resource consuming task.*
+* `run-sql-only` - run all tests in `hazelcast-sql`, `hazelcast-distribution`, and `extensions/mapstore` modules
```
Comment (olukas) [inline README.md:356]: I would remove the word `all` there since we don't run nightly tests by this phrase, right?

**Review by olukas — CHANGES_REQUESTED** [review]
LGTM, just one comment to have it more clear.

File: README.md (hunk 1 above)
Comment (sumnerib) [reply README.md:356]: Right, I should have realized this is what's meant by "all" in the other ones. How about replacing `all` w/ `quick/integration`?

Comment (olukas) [reply README.md:356]: Maybe I would rather use something like `default` for now (there's currently a discussion about how the categories should be named, used etc. so let's keep it as abstract name)

**Review by olukas — APPROVED** [review]

**Review by kwart — APPROVED** [review]

**Pull request merged**